src/components/UI/Modal/Profile/Profile.js

```diff
@@ -0,0 +1,104 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import FormElement from '../../FormElement/FormElement';
+import Button from '../../Button/Button';
+import LinkButton from '../../Button/LinkButton/LinkButton';
+
+import BlankUser from '../../../../assets/images/blank-user.gif';
+
+import * as forms from '../../../../shared/forms';
+import * as actions from '../../../../store/actions/index';
+
+class Profile extends Component {
+  isEdit = this.props.type === 'edit-profile';
+
+  state = {
+    form: {
+      image: {
+        elementType: 'file',
+        elementConfig: {
+          accept: 'image/*'
+        },
+        label: 'Image',
+        value: this.isEdit ? (this.props.photoUrl ? this.props.photoUrl : BlankUser) : BlankUser,
+        validation: {
+          required: false
+        },
+        valid: true
+      },
+      name: {
+        elementType: 'input',
+        elementConfig: {
+          type: 'text',
+          placeholder: 'John Doe'
+        },
+        label: 'Name',
+        value: this.isEdit ? this.props.displayName : '',
+        validation: {
+          required: false
+        },
+        valid: true
+      }
+    }
+  }
+
+  submitProfileForm = (event) => {
+    event.preventDefault();
+
+    this.props.authSetProfile(this.props.token, this.state.form.name.value, this.state.form.image.value, this.isEdit);
+  }
+
+  render() {
+    const { click, loading, error } = this.props;
+
+    const formElementsArray = forms.createFormElementsArray(this.state.form);
+
+    return (
+      <>
+        <h2>My Profile</h2>
+        <form onSubmit={(event) => this.submitProfileForm(event)} className="form">
+          {formElementsArray.map((formElement) => {
```

Comment:
seeing as you don't need to reference the whole `formElement` object here, it's probably more straightforward to implement destructuring:
```suggestion
          {formElementsArray.map(({ id, config }) => {
```

Reply:
I never even thought of that. Fixed it in a few places.